source/opt/ir_loader.cpp

```diff
@@ -90,7 +90,12 @@ bool IrLoader::AddInstruction(const spv_parsed_instruction_t* inst) {
 
   std::unique_ptr<Instruction> spv_inst(
       new Instruction(module()->context(), *inst, std::move(dbg_line_info_)));
-  dbg_line_info_.clear();
+  if (!dbg_line_info_.empty()) {
+    if (dbg_line_info_.back().opcode() == SpvOpNoLine)
+      dbg_line_info_.clear();
+    else
+      dbg_line_info_.erase(dbg_line_info_.begin(), dbg_line_info_.end() - 1);
```

Comment:
I'm not understanding why we're removing the first N - 1 elements of `dbg_line_info_` here.

Reply:
I wanted to keep the last line instruction because it must be applied to the next instruction.
After seeing your comment, I realized that I must keep the last line instruction using a separate variable, not `dbg_line_info_` because processing the new `OpLine` will accumulate the new one in `dbg_line_info_`. It would be a bug. I updated the code to fix it.